### DISCUSSION TITLE
Trigger on page load or on scroll

### DISCUSSION THREAD

**Discussion by 4goodapp** [Flyer53/jsPanel4#166] [Q&A] [answered]
Hi,
Is it possible to show the panel on like 5 second after the page load or maybe after user scroll the page?

**Comment by Flyer53** [comment, marked as answer]
Hi there

Sure, for the page load you could use
```javascript
window.addEventListener('load', (event) => {
  console.log('page is fully loaded');
  jsPanel.create();
});
```

To add a 5 sec delay just wrap the function creating the panel in a setTimeout like
```javascript
window.addEventListener('load', (event) => {
  console.log('page is fully loaded');
  window.setTimeout(function(){
    jsPanel.create();
  }, 5000);
});
```
Take a look at:
https://developer.mozilla.org/en-US/docs/Web/API/Window/load_event
https://developer.mozilla.org/en-US/docs/Web/API/WindowOrWorkerGlobalScope/setTimeout

For scrolling it's basically the same. Look at:
https://developer.mozilla.org/en-US/docs/Web/API/Document/scroll_event
https://developer.mozilla.org/en-US/docs/Web/API/Element/scroll_event

**Reply by 4goodapp** [reply]
Thank you for the response, gonna check this, will let you know.
I want use it for newsletter popup.

**Reply by 4goodapp** [reply]
It works as I expected.
I still have some questions tho.
Like let's suppose I already coded my own popup/modal design with  image and inputs with CSS style, how can I integrate it with jsPanel?
Cause I noticed a basic jsPanel setup as it's own default design with header, title and size predefined.

**Reply by Flyer53** [reply]
There are quite a few options dealing with adding content to the panel. Take a look at:
https://jspanel.de/#options/content
https://jspanel.de/#options/contentAjax (which is the one you should use I think)

Also the header section can be configured using various options where [option header](https://jspanel.de/#options/header) is the most basic one. Please read the docs about all the other available options.

**Reply by 4goodapp** [reply]
Thank you, It's working like I wanted now :)
